lib/sqlalchemy/dialects/mssql/base.py

```diff
@@ -2288,6 +2323,7 @@ class MSDialect(default.DefaultDialect):
     non_native_boolean_check_constraint = False
     supports_unicode_binds = True
     postfetch_lastrowid = True
+    _supports_offset_fetch = False
 
     server_version_info = ()
 
```

Comment:
I believe you also have to omit the usage of TOP in get_select_precolumns(), this is in my patch at https://github.com/sqlalchemy/sqlalchemy/issues/5084#issuecomment-572630940 which I'm assuming this is based off of

Reply:
Usage of TOP is possible without "ORDER BY". If we replace it with OFFSET/FETCH then sorting will be required. 

Moreover you can't use FETCH without OFFSET. This code returns an exception:

SELECT id from users
ORDER BY id
FETCH NEXT 1 ROWS ONLY 

Correct sql is:

SELECT id from users
ORDER BY id
OFFSET 0 ROWS
FETCH NEXT 1 ROWS ONLY 

So I think TOP should remain in cases where user choose simple limit, for example: 
db.Users.query.first()

Reply:
OK that makes sense, as long as we support all combinations.   I am hoping that all combinations are being exercised in https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/testing/suite/test_select.py#L150.   here's a pointless ASCII chart I made just now for no reason:

```
                       offset absent  |   offset simple integer  |  offset sql expression
                      +---------------+--------------------------+------------------------+
limit absent          |_______________|__________________________|________________________|
limit simple integer  |_______________|__________________________|________________________|
limit sql expression  |               |                          |                        |
                      +---------------+--------------------------+------------------------+
```